### PR TITLE
Add razegenie and direct dependencies

### PR DESCRIPTION
**Pull request by arabek**

### CODE REVIEW

File: app-misc/razergenie/razergenie-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+EGIT_REPO_URI="https://github.com/z3ntu/RazerGenie.git"
+EGIT_CLONE_TYPE="shallow"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~amd64-linux"
+
+RDEPEND="
+	app-misc/openrazer[daemon]
+	app-misc/razer-test
```
Comment (vifino) [inline app-misc/razergenie/razergenie-9999.ebuild:20]: Why does this depend on razer_test (supposed to be an openrazer replacement) and on openrazer at the same time?

Comment (arabek) [reply app-misc/razergenie/razergenie-9999.ebuild:20]: Both are available as an option in the gui. Unfortunately the razer_test backend isn't easy to decouple from current Razer Genie build.

Comment (vifino) [reply app-misc/razergenie/razergenie-9999.ebuild:20]: Hmm.. I think we should make openrazer optional via a USE flag, then. Either I can do it after merging or you can.

File: dev-libs/libopenrazer/libopenrazer-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+HOMEPAGE="https://github.com/z3ntu/libopenrazer"
+EGIT_REPO_URI="https://github.com/z3ntu/libopenrazer.git"
+EGIT_CLONE_TYPE="shallow"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~amd64-linux"
+
+RDEPEND="
+	app-misc/razer-test
```
Comment (vifino) [inline dev-libs/libopenrazer/libopenrazer-9999.ebuild:19]: I've looked at `libopenrazer`, it seems to have a dependency on openrazer as well. Not sure if it's optional or required, either way, we should add a USE flag here as well.

File: app-misc/razergenie/razergenie-9999.ebuild (hunk 1 above)
Comment (arabek) [reply app-misc/razergenie/razergenie-9999.ebuild:20]: Making either of those optional will require additional src_prepare trickery as there are no direct build system switches to disable them. I could prepare them, but idk how much time it'll take me and when.

Feel free to take any approach you like.

File: dev-libs/libopenrazer/libopenrazer-9999.ebuild (hunk 2 above)
Comment (arabek) [reply dev-libs/libopenrazer/libopenrazer-9999.ebuild:19]: Good catch!

File: app-misc/razergenie/razergenie-9999.ebuild (hunk 1 above)
Comment (vifino) [reply app-misc/razergenie/razergenie-9999.ebuild:20]: Argh! Yeah, seems like those are actually hard deps.
If they are lacking, it'll fall back to a git clone, I think. So not what we want.

I guess we'll just leave it in for now, see how its behaviour changes in the upstream project.

File: dev-libs/libopenrazer/libopenrazer-9999.ebuild (hunk 2 above)
Comment (vifino) [reply dev-libs/libopenrazer/libopenrazer-9999.ebuild:19]: Unlike razergenie, this does not seem to be a link-time dependency, however I think we should just add a hard dep until we fix razergenie as well.

```suggestion
	app-misc/openrazer[daemon]
	app-misc/razer-test
```

**Comment by arabek** [comment]
I've squashed the commits to keep it clean

**Review by vifino — APPROVED** [review]
Looks great now!

**Pull request merged**